src/n_body.rs

```diff
@@ -68,7 +69,8 @@ struct Planet {
 fn advance(bodies: &mut [Planet;N_BODIES], dt: f64, steps: i32) {
     for _ in 0..steps {
         let mut b_slice: &mut [_] = bodies;
-        while let Some(bi) = shift_mut_ref(&mut b_slice) {
+        while let Some((bi, tail)) = {b_slice}.split_first_mut() {
```

Comment:
Wow, I didn't know this move trick. How are we supposed of discovering that? I like it.